src/backend/InvenTree/report/api.py

```diff
@@ -1,164 +1,314 @@
-                except ValidationError:
-                    continue
 
-                for item in items:
-                    item_query = self.ITEM_MODEL.objects.filter(pk=item.pk)
+class ReportFilter(ReportFilterBase):
+    """Filter class for report template list."""
 
-                    try:
-                        if not item_query.filter(**filters).exists():
-                            matches = False
-                            break
-                    except FieldError:
-                        matches = False
-                        break
+    class Meta:
+        """Filter options."""
 
-                # Matched all items
-                if matches:
-                    valid_report_ids.add(report.pk)
+        model = report.models.ReportTemplate
+        fields = ['landscape']
 
-            # Reduce queryset to only valid matches
-            queryset = queryset.filter(pk__in=list(valid_report_ids))
 
-        return queryset
+class LabelFilter(ReportFilterBase):
+    """Filter class for label template list."""
 
+    class Meta:
+        """Filter options."""
 
-@method_decorator(cache_page(5), name='dispatch')
-class ReportPrintMixin:
-    """Mixin for printing reports."""
+        model = report.models.LabelTemplate
+        fields = []
+
+
+class LabelPrint(GenericAPIView):
+    """API endpoint for printing labels."""
+
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = report.serializers.LabelPrintSerializer
+
+    def get_plugin_class(self, plugin_id: int, raise_error=False):
+        """Return the plugin class for the given plugin key."""
+        from plugin.models import PluginConfig
+
+        plugin = None
+
+        try:
+            plugin_config = PluginConfig.objects.get(pk=plugin_id)
+            plugin = plugin_config.plugin
+        except (ValueError, PluginConfig.DoesNotExist):
+            pass
+
+        error = None
+
+        if not plugin:
+            error = _('Plugin not found')
+        elif not plugin.is_active():
+            error = _('Plugin is not active')
+        elif not plugin.mixin_enabled('labels'):
+            error = _('Plugin does not support label printing')
+
+        if error:
+            plugin = None
+
+            if raise_error:
+                raise ValidationError({'plugin': error})
+
+        return plugin
+
+    def get_plugin_serializer(self, plugin):
+        """Return the serializer for the given plugin."""
+        if plugin and hasattr(plugin, 'get_printing_options_serializer'):
+            return plugin.get_printing_options_serializer(self.request)
+
+        return None
+
+    def get_serializer(self, *args, **kwargs):
+        """Return serializer information for the label print endpoint."""
+        plugin = None
+
+        # Plugin information provided?
+        if self.request:
+            plugin_key = self.request.query_params.get('plugin', None)
```

Comment:
At least this needs to change but I am not sure the `get_plugin_class` fnc is really working as intended
```suggestion
            plugin_key = self.request.data.get('plugin', None)
```

Reply:
I will provide a better tested fix for this in the testing PR